### PR TITLE
HW Oleg Chubenco

### PR DESCRIPTION
**Pull request by 0legCH**

### CODE REVIEW

**Comment by 0legCH** [comment]
HW

File: Inheritance_rhomb/Source.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		this->firm = firm;
+		this->engine_capacity = engine_capacity;
+	}
+	virtual void output(ofstream& os)const
+	{
+		os << "Car name:" << this->name << endl;
+		os << "Brand:" << this->firm << endl;
+		os << "Engine volume:" << this->engine_capacity << endl;
+	}
+	float GetEngine_capacity() {
```
Comment (vtrushevsky) [inline Inheritance_rhomb/Source.cpp:28]: const method

File: Inheritance_rhomb/Source.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		return engine_capacity;
+	}
+	string GetFirm() {
```
Comment (vtrushevsky) [inline Inheritance_rhomb/Source.cpp:31]: const method

File: Inheritance_rhomb/Source.cpp [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+		return firm;
+	}
+	virtual string GetTipe() {
```
Comment (vtrushevsky) [inline Inheritance_rhomb/Source.cpp:34]: const method

File: Inheritance_rhomb/Source.cpp [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+			Toyota[1] = "GasCar";
+		}
+		else if (cars[i]->GetFirm() == "Toyota" && cars[i]->GetTipe() == "HybridCar") {
+			Toyota[2] = "HybridCar";
+		}
+	}
+	//if(Toyota[0] == "ElectroCar" && Toyota[1] == "GasCar" && Toyota[2] == "HybridCar"){ do something}
+	delete[]cars;
+	return 0;
+}
```
Comment (vtrushevsky) [inline Inheritance_rhomb/Source.cpp:152]: Partially done - 4.

File: MultInheritance/Chubenco/Source.cpp [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+		Apothem = 0;
+	}
+	void setData(int Ribs, double Side_length, double Apothem) {
+		this->Ribs = Ribs;
+		this->Heights = Ribs;
+		this->Apothem = Apothem;
+		this->Side_length = Side_length;
+		Square = 0.5 * Apothem * Ribs * Side_length;
+	}
+	double GetSquare() {
```
Comment (vtrushevsky) [inline MultInheritance/Chubenco/Source.cpp:54]: const method

File: MultInheritance/Chubenco/Source.cpp [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+		return Square;
+	}
+	void Print() {
```
Comment (vtrushevsky) [inline MultInheritance/Chubenco/Source.cpp:57]: const method

File: MultInheritance/Chubenco/Source.cpp [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+	cout  << endl;
+	bubbleSort(arr_Volumetric_figure, 5);
+	for (int i = 0; i < 5; i++) {
+		if (arr_Volumetric_figure[i] == c1.getHigh()) { c1.Print(); }
+		else if (arr_Volumetric_figure[i] == c2.getHigh()) { c2.Print(); }
+		else if (arr_Volumetric_figure[i] == c3.getHigh()) { c3.Print(); }
+		else if (arr_Volumetric_figure[i] == d1.getHigh()) { d1.Print(); }
+		else if (arr_Volumetric_figure[i] == d2.getHigh()) { d2.Print(); }
+	}
+}
```
Comment (vtrushevsky) [inline MultInheritance/Chubenco/Source.cpp:208]: 5б

**Pull request closed without merging**